platforms/core-runtime/gradle-cli-main/build.gradle.kts

```diff
@@ -15,14 +15,27 @@
  */
 
 plugins {
-    id("gradlebuild.distribution.api-java")
+    id("gradlebuild.distribution.implementation-java")
+    id("gradlebuild.launchable-jar")
+    id("gradlebuild.start-scripts")
 }
 
-description = "Java 6-compatible entry point of the `gradle` command. See :launcher project for the rest."
+description = "Java 6-compatible entry point of the `gradle` command. Boostraps the Gradle client implementation in :launcher."
 
 gradlebuildJava.usedForStartup()
 
+app {
+    mainClassName = "org.gradle.launcher.GradleMain"
+}
+
 dependencies {
```

Comment:
Also not to do right now, but in the spirit of declarative modeling, do we eventually see the application dependencies being moved into the new `app` block?